packages/account-elements/src/providers/logto-account-provider.ts

```diff
@@ -1,14 +1,54 @@
-import { html, LitElement } from 'lit';
-import { customElement } from 'lit/decorators.js';
+import { createContext, provide } from '@lit/context';
+import type { UserProfileResponse } from '@logto/schemas';
+import { html, LitElement, type PropertyValues } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+import { LogtoAccountApi, type AccessTokenFetcher } from '../api/index.js';
+
+export type LogtoAccountContextType = {
+  api: LogtoAccountApi;
+  userProfile: UserProfileResponse;
+};
+
+export const logtoAccountContext = createContext<LogtoAccountContextType>('logto-account-context');
 
 const tagName = 'logto-account-provider';
-@customElement('logto-account-provider')
+
+@customElement(tagName)
 export class LogtoAccountProvider extends LitElement {
   static tagName = tagName;
 
+  @property({ attribute: 'logto-endpoint' })
+  logtoEndpoint!: string;
+
+  @property({ type: Function })
+  accessTokenFetcher?: AccessTokenFetcher;
```

Comment:
Is this also required?

Reply:
Since we cannot set function or object property in html for web components, so this property should be set in js code, so we make this field optional.
e.g.:
```js
const provider = document.querySelector('logto-account-provider');
provider.accessTokenFetcher = () => { /* ... */}
```

Reply:
If so, how can we strictly guard this field? E.g. throw an error if the required property is missing.

Reply:
It seems difficult to handle because the timing of the user specifying this property is beyond our control. The current solution is that if the user does not specify this value, the `accountContext` cannot be obtained.